code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -77,7 +44,7 @@
 	if (!adjacency_flag)
 		return
 
-	if (is_type_in_list(target, can_be_placed_into))
+	if (target == !splashable())
```

Comment:
Oh, I can't believe I overlooked this before, very amusing

`if(!target.splashable())`

You need to be calling the splashable() proc on the target, not comparing to see if the target is the result of this object's splashable proc

Reply:
I'll fix this when I get home in the evening

Reply:
Fixed!